.github: Remove add_annotations workflow

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64450
* __->__ #64449

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D30738460](https://our.internmc.facebook.com/intern/diff/D30738460)

seemethere committed:
.github: Remove add_annotations workflow

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

suo reviewed (approved):
woooooOOOOO

driazati reviewed (approved):
Can you also get rid of the correpsonding `Upload artifact` steps

seemethere committed:
Update on ".github: Remove add_annotations workflow"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere commented on .github/workflows/lint.yml:
```diff
@@ -318,6 +318,17 @@ jobs:
           fi
+      - name: Add annotations
+        # Don't run on forked pull requests
+        if: ${{ failure() && github.event.pull_request.head.repo.full_name == github.repository }}
```
This will also ensure that this won't run on trunk since `github.event.pull_request.head.repo.full_name` will equal out to `''` if not present

seemethere committed:
Update on ".github: Remove add_annotations workflow"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30738460).

seemethere committed:
Update on ".github: Remove add_annotations workflow"


Signed-off-by: Eli Uriegas <eliuriegasfb.com>

Differential Revision: [D30738460](https://our.internmc.facebook.com/intern/diff/D30738460)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30738460).
